### ISSUE TITLE
trace: Migrate to open-telemetry for instrumentation.

### ISSUE THREAD

**Issue by kavirajk** [weaveworks/common#252]
Currently [Loki](https://github.com/grafana/loki) is instrumented with `opentracing/jaeger` client libraries for tracing (I hope, it's same for Mimir and Tempo as well)

This instrumentation comes from [weaveworks/common](https://github.com/weaveworks/common/blob/master/tracing/tracing.go#L7-L11) package and [dskit's spanlogger](https://github.com/grafana/dskit/blob/main/spanlogger/spanlogger.go#L8-L11) package.

Those client libraries([opentracing](https://github.com/opentracing/specification/issues/163), [jaeger](https://www.jaegertracing.io/docs/1.36/client-libraries/)) are deprecated in the favor of opentelemetry client sdk. It's better to migrate.

I hope, It should be completely possible to migrate underlying dependencies without changing any API of those packages.

Example of using otel tracing client libraries for instrumentation in Go is [here.](https://opentelemetry.io/docs/instrumentation/go/getting-started/#trace-instrumentation)

**Comment by bboreham** [comment]
> It should be completely possible to migrate underlying dependencies without changing any API 

Yeah but not without making tracing stop working for everyone until they figure out how to initialize Otel to make it write to their tracing backend.

Can we put in a layer of indirection?

**Comment by bboreham** [comment]
This came up in conversation today.  Given OpenTelemetry has matured and adoption has increased I'm more OK with a breaking change.

**Comment by DavidSpek** [comment]
@kavirajk Is this something Grafana is going to pick up? It's currently not possible to get spans to propagate to Loki/Mimir/Tempo if OpenTelemetry is used in the stack since it isn't possible to set the [JAEGER_PROPAGATION](https://github.com/jaegertracing/jaeger-client-go/issues/277) environment variable in the Jaeger Go client.

**Comment by ying-jeanne** [comment]
@DavidSpek Hi, thank you for bring up the topic, and yes we are planning to do the migration on our side.
